.github/workflows/test.yml

```diff
@@ -24,40 +18,16 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby:
-          - 2.6
-          - 2.7
-          - "3.0"
-        gemfile:
-          - Gemfile
-          - gemfiles/rack1.gemfile
-          - gemfiles/rack2.gemfile
-          - gemfiles/rack2_2.gemfile
-          - gemfiles/rails_5.gemfile
-          - gemfiles/rails_6.gemfile
-          - gemfiles/rails_6_1.gemfile
-        include:
-          - ruby: 3.1
-            gemfile: 'gemfiles/multi_json.gemfile'
-          - ruby: 3.1
-            gemfile: 'gemfiles/multi_xml.gemfile'
-          - ruby: 3.1
-            gemfile: 'gemfiles/rails_7.gemfile'
-          - ruby: "3.0"
-            gemfile: 'gemfiles/multi_json.gemfile'
-          - ruby: "3.0"
-            gemfile: 'gemfiles/multi_xml.gemfile'
-          - ruby: "3.0"
-            gemfile: 'gemfiles/rails_7.gemfile'
-          - ruby: 2.7
-            gemfile: 'gemfiles/multi_json.gemfile'
-          - ruby: 2.7
-            gemfile: 'gemfiles/multi_xml.gemfile'
-          - ruby: 2.7
-            gemfile: 'gemfiles/rails_7.gemfile'
-    runs-on: ubuntu-20.04
+        os: [ubuntu-latest]
+        ruby: ['2.6', '2.7', '3.0', '3.1', '3.2']
+        gemfile: [multi_json, multi_xml, rack_1_0, rack_2_0, rack_2_2, rails_5_2, rails_6_0, rails_6_1, rails_7_0]
```

Comment:
This becomes a massive matrix and the list of `exclude` will grow.

We could make something like https://github.com/slack-ruby/slack-ruby-client/blob/master/.github/workflows/test.yml#L13 potentially to make it easier to grok, WDYT?

Reply:
I've reduced it and add some specific inclusion.